### DISCUSSION TITLE
DOCX Reader: Is it correct behaviour / best solution to use the last title element present in file?

### DISCUSSION THREAD

**Discussion by Aspvik** [jgm/pandoc#10359] [General]
If there are multiple title elements/styles in a docx file, Pandoc will use the last one. Is this det logical solution? Would it not be better to use the first one?

**Comment by jgm** [comment]
I just tried converting a docx with two elements with the Title style.
Pandoc used the first as the title.  What version of pandoc are you using, and can you upload a docx I can test with?

**Reply by Aspvik** [reply]
Hi!

Thanks for the quick response. First of all we have been using Pandoc for a few years now, and are very grateful for the work you and the maintainers are doing.

Some background: We use Pandoc to convert from docx to epub3, and then post process the chunked xhtml. (Chunked HTML was not an available output when we started using Pandoc, so in the future we should probably use that instead). Our standard command takes advantage of the `--epub-title-page=false` flag, since we don't use that in the post processing, so we have been extracting the document title from the <title> element in the nav.xhtml file.

**Files:**
- single.docx: Single title
- double-empty: Second title element is empty
- double-content: Second title element is non-empty

**Command:**
`pandoc double.docx --from docx --to epub3 -o double.epub`

**Pandoc details:**
pandoc 3.5
Features: +server +lua
Scripting engine: Lua 5.4

**Results**
single.docx
Warning: Non.
nav.xhtml: <title> is set to the first title element.
title_page.xhtml: <title>/\<h1.title> is set to the first title element.

double-empty.docx
Warning: This document format requires a nonempty <title> element. Defaulting to 'double' as the title.
nav.xhtml: <title> is set to the filename.
title_page.xhtml: <title>/\<h1.title> is set to the first title element.

double-content.docx
Warning: This document format requires a nonempty <title> element. Defaulting to 'double' as the title.
nav.xhtml: <title> is set to the filename.
title_page.xhtml:
- <title> is set to a concatenated string of the two titles (First titleSecond title)
- \<h1.title> is the parent of two \<p>, one for each of the two titles.

**Summary**
- There seems to be a mismatch between the <title> element in nav.xhtml and the title used on the title page when there is multiple titles or empty title elements.
- There seems to be a unwanted/unexpected handling of multiple titles (one after the other) in title_page.xhtml where they are concatenated in the <title> and producing non-valid HTML (\<p> child of <\h1>) in the \<body>.

As for my situation I could start converting without the `--epub-title-page=false` flag, and get the title from there instead. However the last case (double-content.docx) prevents this from being a viable option.

[single.docx](https://github.com/user-attachments/files/17675672/single.docx)
[double-empty.docx](https://github.com/user-attachments/files/17675673/double-empty.docx)
[double-content.docx](https://github.com/user-attachments/files/17675674/double-content.docx)

**Reply by jgm** [reply]
Pandoc parses the metadata in double-content.docx as follows:
```
  Meta
    { unMeta =
        fromList
          [ ( "title"
            , MetaBlocks
                [ Para [ Str "First" , Space , Str "title" ]
                , Para [ Str "Second" , Space , Str "title" ]
                ]
            )
          ]
    }
```
So, you have a single title with a two-paragraph content.

In HTML this comes out as
```html
<h1 class="title"><p>First title</p>
<p>Second title</p></h1>
```
which as you observe is malformed.  The HTML writer should do something to ensure that the `title` variable contains only inline content.  It looks like we already do this for the `pagetitle`, and that is why you get the two titles concatenated.  (Perhaps we could do something better in this case.)  It's also interesting that nav.xhtml isn't seeing the title; this is also probably related to it being a two-paragraph block, but I'll have to look into that.

**Reply by jgm** [reply]
Or, we could address this in the docx reader, and ensure that when there are multiple blocks marked "Title", only the first gets used as metadata.

**Reply by Aspvik** [reply]
I don't have any strong opinions on what is the best solution. But it seems to me that addressing the issue in the docx reader and only parse the first title element as the document title would give the most consistent result.

**Comment by jgm** [comment]
I've closed this with a change to the docx reader.
13d37b4713b239169e8a1008952f717d483897d9
